test/common/router/header_formatter_test.cc

```diff
@@ -217,7 +217,9 @@ TEST_F(StreamInfoHeaderFormatterTest, TestFormatWithNonStringPerRequestStateVari
   ON_CALL(stream_info, perRequestState()).WillByDefault(ReturnRef(per_request_state));
   ON_CALL(Const(stream_info), perRequestState()).WillByDefault(ReturnRef(per_request_state));
 
-  testFormatting(stream_info, "PER_REQUEST_STATE(testing)", "");
+  auto f = StreamInfoHeaderFormatter("PER_REQUEST_STATE(testing)", false);
+  EXPECT_THROW_WITH_MESSAGE(f.format(stream_info), EnvoyException,
```

Comment:
Can you explain why this behavior changed?

Reply:
Side effect of removing the template stuff from hasData. Code now has to have try catch stuff around getData. the old code assumed that if an exception is thrown in hasData<string..> called in the StreamInfoHeaderFormatter constructor, then set the field entry to "". It does not apply anymore. The exception is now thrown during the formatting (f.format)